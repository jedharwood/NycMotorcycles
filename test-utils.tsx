import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { IntlProvider } from 'react-intl';
import en from './languages/en.json';
import { MemoryRouterProvider } from 'next-router-mock/MemoryRouterProvider';
import { QueryClient, QueryClientProvider } from 'react-query';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  const queryClient = new QueryClient();

  return (
    <MemoryRouterProvider>
      <QueryClientProvider client={queryClient}>
        <IntlProvider locale='en' messages={en}>
          {children}
        </IntlProvider>
      </QueryClientProvider>
    </MemoryRouterProvider>
  );
};

const renderWithProviders = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { renderWithProviders as render };
