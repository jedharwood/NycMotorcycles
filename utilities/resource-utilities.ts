export const yahooAuctionProfilePage: string =
  'https://auctions.yahoo.co.jp/seller/lazzamoto?'

export const instagramProfilePage: string =
  'https://www.instagram.com/newyorkcitymotorcycles/'

export const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
}

const gallery: string = '/sold-archive/gallery/'
export const galleryRoutes: { [key: string]: string } = {
  hondaRc30: `${gallery}honda-rc30`,
  suzukiRg400: `${gallery}suzuki-rg400`,
  shovelheadChopper: `${gallery}shovelhead-chopper`,
  laverda750Sfc: `${gallery}laverda-750-sfc`,
  harleyXr750: `${gallery}harley-xr750`,
  kawasakiH2: `${gallery}kawasaki-h2`,
  suzukiStinger: `${gallery}suzuki-stinger`,
  bsaStarfire: `${gallery}bsa-starfire`,
  kawasakiH1: `${gallery}kawasaki-h1`,
}
