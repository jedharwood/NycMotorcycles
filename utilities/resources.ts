export const instagramLink: string =
    'https://www.instagram.com/newyorkcitymotorcycles/';
