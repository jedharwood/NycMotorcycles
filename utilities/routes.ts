const routes: { [key: string]: string } = {
  home: '/',
  activeAuctions: '/active-auctions',
  soldArchive: '/sold-archive',
  history: '/history',
  racing: '/racing',
  contact: '/contact',
  consign: '/consign',
  invest: '/invest',
  download: '/download',
  gallery: '/sold-archive/gallery',
};

export default routes;
