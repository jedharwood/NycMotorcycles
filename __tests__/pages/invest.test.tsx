import userEvent from '@testing-library/user-event';
import mockRouter from 'next-router-mock';

import InvestPage from '@/pages/invest';
import { render, screen, waitFor, act } from '@/test-utils';
import { langs } from '@/utilities/resources';

jest.mock('next/router', () => jest.requireActual('next-router-mock'));

describe('InvestPage', () => {
    it('should render page in English', () => {
        const { container } = render(<InvestPage />);

        expect(container).toMatchSnapshot();
    });

    it('should render page in Japanese', () => {
        const { container } = render(<InvestPage />, { locale: langs.ja });

        expect(container).toMatchSnapshot();
    });

    it('should navigate to contact page when Contact button is clicked', async () => {
        const user = userEvent.setup();
        render(<InvestPage />);

        await expect(mockRouter.asPath).toEqual('/');

        const contactButton: HTMLElement = screen.getByTestId(
            'invest-page-contact-button'
        );

        await act(() => {
            user.click(contactButton);
        });

        await waitFor(() => {
            expect(mockRouter.asPath).toEqual('/contact');
        });
    });
});
