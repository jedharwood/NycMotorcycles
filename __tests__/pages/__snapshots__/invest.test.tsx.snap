// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestPage should render page in English 1`] = `
<div>
  <main>
    <div
      class="grid gap-4 md:grid-cols-2"
    >
      <section
        class="space-y-4"
        data-testid="image-grid"
      >
        <div
          class="rounded-md
            cursor-default shadow-lg 
            h-fit"
          role="button"
        >
          <img
            alt="A photograph of a silver coloured vintage motorcycle being displayed in Sotheby's auction house."
          />
        </div>
      </section>
      <div
        class="space-y-4"
      >
        <section
          class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
        >
          <div
            class="text-center "
          >
            <h2
              class="text-xl font-medium opacity-80 md:text-2xl"
            >
              Investment
            </h2>
          </div>
          <div
            class="space-y-4 undefined"
          >
            <p>
              As a portfolio investment, motorcycles as an asset class, professionally-sourced and possessing meaningful character and provenance, outperform the S & P 500 year after year.
            </p>
            <p>
              Demand for coveted motorcycles far outstrips supply; this puts significant upward pressure on prices. Tracking similar trends to four wheeled automobiles, demand has increased exponentially during the recent decade.
            </p>
            <p>
              Two reasons: Social media = envy. That drives those with the means to step up their game. In the past trophies were kept inside a cabinet or on the mantle. Now its on your instagram and twitter.
            </p>
            <p>
              The second reason is the art market. Investment art has also consistently outperformed many financial markets. In the past 10 years, art buyers - not just car and motorcycle collectors have been responsible for driving prices ever higher. Think about it. Why spend (invest) $100K for a static sculpture. The symphony of hundreds, thousands of moving parts...motion, contours, aesthetics. Moving scupture. Like collectible cars, motorcycles as sound investment are getting the respect - and promotion - from unlikely places. Art dealers, writers and critics.
            </p>
            <p>
              New York City Motorcycles acts as a buyer's agent. As an independent third party with deep knowledge and contacts in the vintage motorcycle world, we never accept what is claimed to be a motorcycle's history and provenance. We prove it. We make sure that prices defend value. These are motorcycles that are intended to be liquid, desirable and offer conisistent high growth at the lowest possible risk. Call or email for further guidance.
            </p>
          </div>
          <div
            class="flex justify-center"
          >
            <a
              data-testid="invest-page-contact-button"
              href="/contact"
            >
              <div
                class="flex justify-center"
              >
                <button
                  class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
                  type="button"
                >
                  Contact
                </button>
              </div>
            </a>
          </div>
        </section>
      </div>
    </div>
  </main>
</div>
`;

exports[`InvestPage should render page in Japanese 1`] = `
<div>
  <main>
    <div
      class="grid gap-4 md:grid-cols-2"
    >
      <section
        class="space-y-4"
        data-testid="image-grid"
      >
        <div
          class="rounded-md
            cursor-default shadow-lg 
            h-fit"
          role="button"
        >
          <img
            alt="placeholder"
          />
        </div>
      </section>
      <div
        class="space-y-4"
      >
        <section
          class="bg-stone-600 w-full py-4 px-6 rounded-md text-stone-50 shadow-lg space-y-4 bg-opacity-90 undefined"
        >
          <div
            class="text-center "
          >
            <h2
              class="text-xl font-medium opacity-80 md:text-2xl"
            >
              placeholder
            </h2>
          </div>
          <div
            class="space-y-4 undefined"
          >
            <p>
              placeholder
            </p>
            <p>
              placeholder
            </p>
            <p>
              placeholder
            </p>
            <p>
              placeholder
            </p>
            <p>
              placeholder
            </p>
          </div>
          <div
            class="flex justify-center"
          >
            <a
              data-testid="invest-page-contact-button"
              href="/contact"
            >
              <div
                class="flex justify-center"
              >
                <button
                  class="rounded-md px-6 py-2 text-lg bg-teal-700 hover:bg-teal-500"
                  type="button"
                >
                  placeholder
                </button>
              </div>
            </a>
          </div>
        </section>
      </div>
    </div>
  </main>
</div>
`;
